src/optimization/LiteralValues.cpp

```diff
@@ -791,7 +791,6 @@ InstructionWalker optimizations::handleUseWithImmediate(
                     // and with constant/unsigned, etc.)
                     if(localIt->getSingleWriter() != nullptr)
                         it->addDecorations(intermediate::forwardDecorations(localIt->getSingleWriter()->decoration));
-                    it.nextInBlock();
                     op->replaceLocal(oldLocal, tmp.local, LocalUse::Type::READER);
```

Comment:
Why did you remove this instruction here? Was it a bug?

Reply:
This is a bug. Fixed.